### PR TITLE
Make postinstall exit on error + other postinstall fixes

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue
- This PR fixes the most glaring issues in our Traffic Ops post-install scripts, including not exiting on error, mangling backslashes, and unglobbing passwords.
- This PR makes it possible to run the `postinstall` Bash script in CDN-in-a-Box

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

1. Rebuild Traffic Ops Perl and start CDN-in-a-Box
2. <details><summary>Build CDN-in-a-Box and run these commands (click to expand)</summary>

    ```shell script
    <<'JSON_INPUT_FILE' cat > configuration_file.json;
    {
      "/opt/traffic_ops/app/conf/cdn.conf": [
        {
          "Generate a new secret?": "yes",
          "config_var": "genSecret"
        },
        {
          "Number of secrets to keep?": "2",
          "config_var": "keepSecrets"
        },
        {
          "Number of workers?": "12",
          "config_var": "workers"
        },
        {
          "Traffic Ops url?": "http://trafficops.infra.ciab.test:3000",
          "config_var": "base_url"
        }
      ],
      "/opt/traffic_ops/app/conf/ldap.conf": [
        {
          "Do you want to set up LDAP?": "no",
          "config_var": "setupLdap"
        }
      ],
      "/opt/traffic_ops/app/conf/production/database.conf": [
        {
          "Database type": "Pg",
          "config_var": "type"
        },
        {
          "Database name": "traffic_ops",
          "config_var": "dbname"
        },
        {
          "Database server hostname IP or FQDN": "db.infra.ciab.test",
          "config_var": "hostname"
        },
        {
          "Database port number": "5432",
          "config_var": "port"
        },
        {
          "Traffic Ops database user": "traffic_ops",
          "config_var": "user"
        },
        {
          "Traffic Ops database password": "twelve",
          "config_var": "password",
          "hidden": "1"
        }
      ],
      "/opt/traffic_ops/app/db/dbconf.yml": [
        {
          "Database server root (admin) username": "traffic_ops",
          "config_var": "pgUser"
        },
        {
          "Database server admin password": "twelve",
          "config_var": "pgPassword",
          "hidden": "1"
        },
        {
          "Download Maxmind Database?": "no",
          "config_var": "maxmind"
        }
      ],
      "/opt/traffic_ops/install/data/json/openssl_configuration.json": [
        {
          "Do you want to generate a certificate?": "yes",
          "config_var": "genCert"
        },
        {
          "Country Name (2 letter code)": "US",
          "config_var": "country"
        },
        {
          "State or Province Name (full name)": "CO",
          "config_var": "state"
        },
        {
          "Locality Name (eg, city)": "Denver",
          "config_var": "locality"
        },
        {
          "Organization Name (eg, company)": "Comcast",
          "config_var": "company"
        },
        {
          "Organizational Unit Name (eg, section)": "Traffic Control",
          "config_var": "org_unit"
        },
        {
          "Common Name (eg, your name or your server's hostname)": "trafficops.infra.ciab.test",
          "config_var": "common_name"
        },
        {
          "RSA Passphrase": "rsapass",
          "config_var": "rsaPassword",
          "hidden": "1"
        }
      ],
      "/opt/traffic_ops/install/data/json/profiles.json": [
        {
          "Traffic Ops url": "https://trafficops.infra.ciab.test",
          "config_var": "tm.url"
        },
        {
          "Human-readable CDN Name.  (No whitespace, please)": "CDN-in-a-Box",
          "config_var": "cdn_name"
        },
        {
          "DNS sub-domain for which your CDN is authoritative": "mycdn.ciab.test",
          "config_var": "dns_subdomain"
        }
      ],
      "/opt/traffic_ops/install/data/json/users.json": [
        {
          "Administration username for Traffic Ops": "admin",
          "config_var": "tmAdminUser"
        },
        {
          "Password for the admin user": "twelve",
          "config_var": "tmAdminPw",
          "hidden": "1"
        }
      ],
      "/opt/traffic_ops/install/data/profiles/": [
        {
          "Add custom profiles?": "no",
          "config_var": "custom_profiles"
        }
      ]
    }
    JSON_INPUT_FILE
    docker cp configuration_file.json $(docker-compose ps -q trafficops-perl):/opt/traffic_ops/app/;
    docker-compose exec trafficops-perl ../install/bin/postinstall -a -cfile configuration_file.json;
    ```
   </details>

In the future, we should make the `postinstall` script run in CDN-in-a-Box. 

## If this is a bug fix, what versions of Traffic Control are affected?

- master (46fad538bb)
- 4.0.0
- 4.1.0 (RC0)

## The following criteria are ALL met by this PR

- [x] This PR includes a test in the *What is the best way to verify this PR?* section.
- [x] Intended behavior does not change, documentation is unnecessary
- [x] An update CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/install/bin/postinstall [hunk 1]
```diff
@@ -25,7 +28,7 @@ done
 
 # above options don't require any of the extra processing -- just kick off the script with
 # options intact.
-if [[ $action == "bypass" ]]; then
+if [[ -v action && "$action" == "bypass" ]]; then
```
Comment (rawlinp) [inline traffic_ops/install/bin/postinstall:31]: What does `-v` do here?

Comment (zrhoffman) [reply traffic_ops/install/bin/postinstall:31]: `-v` checks if `action` is defined without expanding it. It's generally used in combination with `set -o nounset` (`set -u`).

Comment (rawlinp) [reply traffic_ops/install/bin/postinstall:31]: ah, thanks, I couldn't find it in the ancient version of bash that macOS includes by default

File: traffic_ops/install/bin/install_goose.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-trap "echo Error on line ${LINENO} of '$0'; exit 1" ERR
+trap 'echo "Error on line ${LINENO} of ${0}"; exit 1' ERR
```
Comment (ocket8888) [inline traffic_ops/install/bin/install_goose.sh:17]: Why?

Comment (zrhoffman) [reply traffic_ops/install/bin/install_goose.sh:17]: `$LINENO` and `$0` are supposed to be expanded when a non-conditional line returns a nonzero exit code, which evaluates the ERR signal trap. Before, they were being set when the script started.

Comment (ocket8888) [reply traffic_ops/install/bin/install_goose.sh:17]: Was `LINENO`? because `$0` shouldn't change during script execution

Comment (zrhoffman) [reply traffic_ops/install/bin/install_goose.sh:17]: Yep $LINENO is the current line that is executing, and to get an accurate value in the trap, it needs to expand no sooner than a line returns an error.

Comment (ocket8888) [reply traffic_ops/install/bin/install_goose.sh:17]: Alright, I'll take your word for it

**Pull request merged**